lib/board.rb

```diff
@@ -76,16 +76,29 @@ def ship_number_coords_valid?(ship, coordinates)
   def ship_diagonal_coords_valid?(ship, coordinates)
     (((ship_number_coords_valid?(ship, coordinates)) &&
     (ship_letter_coords_valid?(ship, coordinates))) &&
-    (coordinates_ords(coordinates).count == 1))
+    (coordinates_ords(coordinates).count == 1) ||
+    ((ship_number_coords_valid?(ship, coordinates)) &&
+    (ship_letter_coords_valid?(ship, coordinates))) &&
+    (coordinates_nums(coordinates).count == 1))
```

Comment:
We are doing great! Let's make some time to refactor some methods today or tomorrow.